lib/store.js

```diff
@@ -40,16 +59,29 @@ Store.prototype.create = function(modelName, obj) {
 	var that = this;
 	var model = this.getModel(modelName);
 	return model.schema.create(obj).then(function(instance) {
-		return that._set(model, instance, 'create');
+		var promise = that._set(model, instance, 'create');
+		var pk = model.schema.getPkOfInstance(instance);
+		if (that.cacheEnabled && pk !== null) {
+			var key = cacheKey(modelName, pk);
+			that.cache.set(key, promise);
+		}
+		return promise;
 	});
 };
 
 Store.prototype.update = function(modelName, obj) {
 	obj = obj || {};
 	var that = this;
 	var model = this.getModel(modelName);
+
 	return model.schema.create(obj).then(function(instance) {
-		return that._set(model, instance, 'update');
+		var promise = that._set(model, instance, 'update');
```

Comment:
[Low] This piece of code is almost identical to the one above. We can refactor it (given we can distinguish between `create` and `update`.